daemon/daemon.go

```diff
@@ -407,20 +407,12 @@ func (daemon *Daemon) reserveName(id, name string) (string, error) {
 
 		conflictingContainer, err := daemon.GetByName(name)
 		if err != nil {
-			if strings.Contains(err.Error(), "Could not find entity") {
-				return "", err
-			}
```

Comment:
I came across a condition where the container record exists in the sqlite database (aka `containerGraphDB` to the daemon) but absent from the `contStore`.
I switched the storage driver from device mapper to aufs long time ago. The containers that were created with device mapper couldn't be registered to the `contStore` because of different storage drivers.
In this case when I run `docker ps -a` it shows no containers.
But if I try to create a container with the name that were used by the broken container in the sqlite database,  the `err` here is ` Could not find container for entity id xxx` and the create command fails with `Error response from daemon: Could not find container for entity id f2bd2b291853fe70c9b3cc378d2bbf04c2089d5c61893748761bcbfc795c5911`. This is really confusing.

I think the `if` condition above is opposite and should be changed to 
```go
if !strings.Contains(err.Error(), "Could not find entity") {
	return "", err
} else {
	// Remove name and continue starting the container
}
```

If it's unsafe to do this, I think we should either refuse the container to start or delete the broken container records from database at start up if some of the containers were created with different storage drivers.

My 0.02 cents :smile_cat: